### PR TITLE
Pool: parse rlp chain id for non-legacy transactions

### PR DESCRIPTION
**Pull request by AskAlexSharov**

### CODE REVIEW

File: types/txn.go [hunk 1]
```diff
@@ -182,13 +182,17 @@ func (ctx *TxParseContext) ParseTransaction(payload []byte, pos int, slot *TxSlo
...
 	sigHashPos := p
-	// If it is non-legacy tx, chainId follows, but we skip it
 	if !legacy {
-		dataPos, dataLen, err = rlp.String(payload, p)
+		p, err = rlp.U256(payload, p, &ctx.ChainID)
 		if err != nil {
 			return 0, fmt.Errorf("%w: chainId len: %s", ErrParseTxn, err)
 		}
-		p = dataPos + dataLen
+		if ctx.ChainID.IsZero() { // zero indicates that the chain ID was not specified in the tx.
```
Comment (yperbasis) [inline types/txn.go:190]: I don't think we should allow 0 chain ID for non-legacy transactions. I believe geth disallows 0 chain ID in their transaction pool even for legacy transactions and only allows pre-[EIP-155](https://eips.ethereum.org/EIPS/eip-155) transactions in incoming blocks assembled by other clients.

**Review by yperbasis — CHANGES_REQUESTED** [review]

File: types/txn.go (hunk 1 above)
Comment (AskAlexSharov) [reply types/txn.go:190]: @yperbasis i referred to this comment:  https://github.com/ethereum/go-ethereum/blob/c503f98f6d5e80e079c1d8a3601d188af2a899da/core/types/transaction_signing.go#L208

Comment (yperbasis) [reply types/txn.go:190]: > @yperbasis i referred to this comment: https://github.com/ethereum/go-ethereum/blob/c503f98f6d5e80e079c1d8a3601d188af2a899da/core/types/transaction_signing.go#L208

Yeah, but in their txpool they use `(s londonSigner) Sender`, which doesn't accept 0 chain ID, not `(s londonSigner) SignatureValues`.

File: types/txn.go [hunk 2]
```diff
@@ -113,6 +114,10 @@ var ErrRlpTooBig = errors.New("txn rlp too big")
 
 func (ctx *TxParseContext) ValidateRLP(f func(txnRlp []byte) error) { ctx.validateRlp = f }
 func (ctx *TxParseContext) WithSender(v bool)                       { ctx.withSender = v }
+func (ctx *TxParseContext) ChainIDReuired() *TxParseContext {
```
Comment (yperbasis) [inline types/txn.go:117]: typo in "Reuired"

**Pull request merged**